### PR TITLE
Fixes #2222 URLDownload now resolves redirects by default

### LINKED ISSUE #2222: Look up full text ArXiv broken
The functionality added in https://github.com/JabRef/jabref/issues/1023 is broken in the current build. Namely it fails to fetch the pdf file. It generates an empty file without extension. 

Also by default when adding an entry from the ArXiv web search it includes the online link. I would like to deactivate that or a way to order that last with respect to the local PDF copy.

### PR DESCRIPTION
**Pull request by stefan-kolb**

### CODE REVIEW

File: src/main/java/net/sf/jabref/logic/net/URLDownload.java [hunk 1]
```diff
@@ -119,6 +118,20 @@ private URLConnection openConnection() throws IOException {
...
+        // normally, 3xx is redirect
+        int status = connection.getResponseCode();
+        if (status != HttpURLConnection.HTTP_OK) {
+            if (status == HttpURLConnection.HTTP_MOVED_TEMP
+                    || status == HttpURLConnection.HTTP_MOVED_PERM
+                    || status == HttpURLConnection.HTTP_SEE_OTHER) {
+                // get redirect url from "location" header field
+                String newUrl = connection.getHeaderField("Location");
+                // open the new connnection again
+                connection = (HttpURLConnection) new URLDownload(newUrl).openConnection();
```
Comment (koppor) [inline src/main/java/net/sf/jabref/logic/net/URLDownload.java]: Can this be tested with mockito?

Can't it happen that the new location also points to a new location?

The client library doesn't transparently redirects and there is no utility layer in the lib? Strange....

Comment (Siedlerchr) [inline src/main/java/net/sf/jabref/logic/net/URLDownload.java]: It sounds weird, but it is the truth!
Explaination from the sun java engineers: (Security Reasons)
http://bugs.java.com/bugdatabase/view_bug.do?bug_id=4620571

Comment (Siedlerchr) [inline src/main/java/net/sf/jabref/logic/net/URLDownload.java]: Addtionally we should nontheless add conn.setInstanceFollowRedirects(true);

**Comment by tobiasdiez** [comment]
Could you please also add a test for the arxiv entry which failed to download before the fix.

File: src/main/java/net/sf/jabref/logic/net/URLDownload.java (hunk 1 above)
Comment (stefan-kolb) [inline src/main/java/net/sf/jabref/logic/net/URLDownload.java]: I think this is the default!

**Comment by stefan-kolb** [comment]
A test for the entry would be no different. It must be done for the URLDownloader, but I'm not entirely sure how right now.

**Pull request merged**